src/components/content/donations-banner-experiment/donations-banner-inline.tsx

```diff
@@ -0,0 +1,194 @@
+        </div>
+      ),
+      call: 'Wusstest du schon …',
+      imageSrc: '/_assets/img/donations/donation-bird.svg',
+    },
+    {
+      id: 'banner-exercisefolder-B2',
+      isLong: false,
+      text: (
+        <div className="text-left">
+          <p className="serlo-p special-hyphens-initial leading-6">
+            … dass wir alle Inhalte auf Serlo ehrenamtlich erstellen? Hier
+            engagieren sich hunderte Lehrkräfte, Lehramtsstudierende und
+            Pädagog*innen für das gemeinsame Ziel:{' '}
+            <b>
+              Kostenloser Zugang zu hochwertigen Lernmaterialien – für alle
+              Schüler*innen
+            </b>
+            .
+          </p>
+          <p className="serlo-p special-hyphens-initial leading-6">
+            Als nächstes möchten wir gerne weitere Übungsformate für Serlo
+            entwickeln, die Nutzbarkeit auf Smartphones verbessern und es
+            Schüler*innen ermöglichen, eigene Lernziele zu definieren und ihren
+            eigenen Lernstand zu speichern.
+          </p>
+          <p className="serlo-p special-hyphens-initial leading-6 font-bold">
+            Unterstütze uns, Serlo noch besser zu machen!
+          </p>
+        </div>
+      ),
+      call: 'Wusstest du schon …',
+      imageSrc: '/_assets/img/donations/donation-bird.svg',
+    },
+  ],
+}
+
+interface Banner {
+  id: string
+  isLong: boolean
+  text: React.ReactNode
+  call: string
+  imageSrc: `/_assets/img/donations/${string}.svg`
+}
+
+export interface DonationsBannerProps {
+  id: number
+  place: 'solution' | 'exercise'
+}
+
+export function DonationsBannerInline({ id, place }: DonationsBannerProps) {
+  const [banner, setBanner] = useState<Banner | undefined>(undefined)
+  const bannerRef = useRef<HTMLElement>(null)
+  const router = useRouter()
+  const { lang } = useInstanceData()
+
+  useEffect(() => {
+    globalThis.hack__id = id
+
+    if (lang !== Instance.De || Math.random() > reducedChance) {
+      setBanner(undefined)
+      return
+    }
+
+    const banners = possibleBanners[place]
+    const chosenBannerIndex = Math.floor(Math.random() * banners.length)
+    setBanner(banners.at(chosenBannerIndex))
+
+    const horizon = document.getElementById('horizon')
+    if (horizon) horizon.style.display = 'none'
+
+    // rerole on entity change
+  }, [setBanner, id, lang, place])
+
+  if (banner == null) return null
+
+  return (
+    <Lazy slim>
+      <aside
+        ref={bannerRef}
+        onLoad={() => {
+          const observer = new IntersectionObserver(
+            (entries) => {
+              const entry = entries.at(0)
+              if (
+                entry &&
+                entry.isIntersecting &&
+                globalThis.hack__lastId !== globalThis.hack__id
+              ) {
+                globalThis.hack__lastId = globalThis.hack__id
+                submitEvent(`spenden-seen-${banner.id}`)
+              }
+            },
+            { threshold: 0.8 }
```

Comment:
BTW: Here you can play with the threshold (before an event was send when the first pixel of the banner was visible). Know you get an event, every time the threshold is crossed (more than 80% of the banner becomes visible / less than 80% becomes visible)